lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -121,13 +128,80 @@ class CombineSelectMaskedLoadPattern : public RewritePattern {
     if (!mask)
       return failure();
 
-    auto splatOp = mask.getDefiningOp<SplatOp>();
-    if (!splatOp)
-      return failure();
+    bool findTargetPattern = false;
+    if (mask == condSelect) {
+      // Case 1: The mask of the load and the cond of the select are the same
+      // value.
+      // ```mlir
+      //   %mask : i1
+      //   %true_val = tt.load %ptr, %mask, %other : !tt.ptr<f32>
+      //   %res = arith.select %mask, %true_val, %false_val : f32
+      // ```
+      findTargetPattern = true;
+    } else if (auto splatOp = mask.getDefiningOp<SplatOp>()) {
+      // Case 2: The mask of the load is splatted from the cond of the select.
+      // ```mlir
+      //   %mask = tt.splat %cond : i1 -> tensor<8xi1>
```

Comment:
this IR should get canonicalized to the first form instead if checking all the different form of IR

Reply:
Case 2 is the scenario originally supported by the `CombineSelectMaskedLoadPattern`, and it won’t be canonicalized. 

The test IR for Case 1 is shown below. The previous pattern didn’t check whether the mask of tt.load and the condition of arith.select were the same.

```mlir
tt.func @test_combine_select_masked_load_pattern1(%ptr: !tt.ptr<f32>, %cond: i1) -> (f32) {
    %false_val = arith.constant 0.0 : f32
    %load = tt.load %ptr, %cond : !tt.ptr<f32>
    %res = arith.select %cond, %load, %false_val : f32
    tt.return %res : f32
}
```